server/application-server/src/main/java/de/tum/in/www1/hephaestus/codereview/comment/IssueComment.java

```diff
@@ -0,0 +1,49 @@
+package de.tum.in.www1.hephaestus.codereview.comment;
+
+import jakarta.persistence.Table;
+
+import org.springframework.lang.NonNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.hephaestus.codereview.base.BaseGitServiceEntity;
+import de.tum.in.www1.hephaestus.codereview.pullrequest.PullRequest;
+import de.tum.in.www1.hephaestus.codereview.user.User;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Table(name = "issue_comment")
+@Getter
+@Setter
+@NoArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+@ToString(callSuper = true)
+public class IssueComment extends BaseGitServiceEntity {
+    @NonNull
+    private String body;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "author_id")
+    @ToString.Exclude
+    private User author;
+
+    @ManyToOne(optional = false)
+    @JoinColumn(name = "pullrequest_id", referencedColumnName = "id")
+    @JsonIgnore
+    @ToString.Exclude
+    private PullRequest pullrequest;
+
+    public IssueComment(String body, String createdAt, String updatedAt) {
```

Comment:
The ID is auto-generated, right? So we don't need it in the constructor.
I will give `@AllArgsConstructor` a try, didn't know that existed.

Reply:
I see, actually, I think we need to remove `@GeneratedValue(strategy = GenerationType.IDENTITY)` for all GitServiceEntities, now I understand it. We would be using the ID that comes from GitHub and not generate it on our own.

Reply:
Oh, so you want to use Github's ID instead? I thought we would later on just add a `provider_id` variable. I am personally not sure if using Github's ID is a good idea since there is no guarantee these are unique across providers.

Reply:
Thank you for clarifying! That makes way more sense.

I also gave `@AllArgsConstructor` a shot. Unfortunately, it seems to be a known issue that it does not allow to initialize attributes of superclasses, so we will have to create the constructors ourselves. See [lombok#2624](https://github.com/projectlombok/lombok/issues/2646).